### ISSUE TITLE
Flow types are incompatible with Flow versions <= 0.45

### ISSUE THREAD

**Issue by fc** [then/promise#135]
It's killing our builds:

```
node_modules/promise/index.js.flow:3
  3: declare class ThenPromise<+R> extends Promise<R> {
                                ^ R. This type is incompatible with
 14:   catch<U>(
       ^ some incompatible instantiation of `U`

node_modules/promise/index.js.flow:14
 14:   catch<U>(
       ^ some incompatible instantiation of `U`. This type is incompatible with
639:       onReject?: (error: any) => ?Promise<U> | U
                                      ^^^^^^^^^^^ null. See lib: /tmp/flow/flowlib_423a7f2/core.js:639

node_modules/promise/index.js.flow:14
 14:   catch<U>(
       ^ some incompatible instantiation of `U`. This type is incompatible with
639:       onReject?: (error: any) => ?Promise<U> | U
                                      ^^^^^^^^^^^ undefined. See lib: /tmp/flow/flowlib_423a7f2/core.js:639
```
This is with flow 0.42.0 which appears to be the cause.

Adding this to `.flowconfig` works around it:
```
[ignore]
.*/node_modules/promise/
```

You can verify this for yourself through Flow's website (just change the dropdown to 0.45.0).
https://flow.org/try/#0PTAEAEDMBsHsHcBQiAmBTAxtAhgJzaFtgM7GgAqAFmgHYAKusAtgJbFoA8A1AEoB8oNAA8ALrRRkGzNp36gA3olCFYNYiNwBXDCNi4AFBmzRoAI2wYA1gC5Q+pctD5isaADc0t-c83QRtqVZ2DjkAH1AeAEpQAF4BN1gWFAAaB2V8ACtMf2V9NFxGXFtsGgBPaLjQBKSHCoFWITQUSNtqlABuZGURahoOAFU+e0dQVQAxX0gWEwB+LzdjTU8IutBAmQGBcP7UkdUeNCydObt8wuKy1fXgwdBt2tsqWmvOQc6HIxEMSk3hx33DtkTnkCnoLuVYgIXps7qB+g8KL1oTxYW8uqAQKAAKKiWjEFiqMjEAAOmBYUwwoF0VN6wGJjCCaHRwAAVCy0izQABBEQiCzUMhGEzmKxkSB6GkEZyuTQiAk0UAlFDACWZbLy0aQSWI54MmTJUDwFg9WCypxoESaXA0Fg0ADmitANDQ8FA9Ok7AAdByIMS8NgmKMaJBJtNoE0dYRjGYLJYqbBBI0MLKCPBetqnvQ9exQGxzS53E1vY5OeA-bgA0G1ToI08o8LY-HE5gU4b0z0CJmXrmyNWxChi8oWcAHChVGh9KoQ9ApiYmsCFtAlrYopDFWUDao+-OvPgSKpwasSuVWokOsgHKz2SXQABhaNkbBO2DoT0ZIkiUrh+sxqye5QAJJajQ449m6jBuEkTQGh24EegQeb4Ja1pFhyI7KCB6DkqUhjRiKNhVGesI0L40AtDqWbwSEfCdBhL5oNhuENlYXhnLYWKgrgBqLsuKxrm05FtO8yjqNgcqUtKhYcOQQywKYRw5NCMmwuQ5FdtmnAybRoCieJ5oKdJQxnHoJzHmpSIaYZ2m6SwlLRhwWLhkwBrkNYAFiBWpjhg5Tl8EM7qMsQjzmbqVEACTkJoxLhgAstgxLSTBpSkrAWphdg8DYMafnWXyekVhgmkGo5aBMAElnKeEMn6AFMhBdyBTYKUPmlXwIWUYyVnMmAOJiGo8pEqSGDkrZTYdjQdIaeiNmUugmEMZAOGQDQtgTDQOjyuR+iejteB2vVx4ANoALqrOpVHHjRDgzc+WGLfoy2rZo61yqo5FrRtqidAAvogQA

**Comment by dkfiresky** [comment]
@fc thanks for your solution.

**Comment by ForbesLindesay** [comment]
Thanks for reporting and for the clear reproduction.  I've released 7.3.1 with the flow types removed, then released them as a breaking release in 8.0.0.  Please ensure you update typescript before updating to the latest version of promise.
